pkg/adapter/nimbus-kyverno/processor/kpbuilder.go

```diff
@@ -111,15 +214,23 @@ func cocoRuntimeAddition(np *v1alpha1.NimbusPolicy, rule v1alpha1.Rule) kyvernov
 		},
 	}
 
-	if len(labels) > 0 {
-		kp.Spec.Rules[0].MatchResources.Any[0].ResourceDescription.Selector.MatchLabels = labels
+	mutateNonExistingKp.Name = np.Name + "-mutateoncreate"
+
+	if (len(deployNames) > 0) || (len(labels) == 0 && len(deployments.Items) > 0)  {  // if labels are present but no deploy exists with matching label or labels are not present but deployments exists
```

Comment:
Not clear why we need the case when len(deployNames) = 0.

Reply:
deployNames is populated only if the len(labels) > 0. If len(labels) is zero, then we need to look at the deployment items